sympy/simplify/tests/test_radsimp.py

```diff
@@ -420,3 +420,15 @@ def test_issue_5933():
     x = Polygon(*RegularPolygon((0, 0), 1, 5).vertices).centroid.x
     assert abs(denom(x).n()) > 1e-12
     assert abs(denom(radsimp(x))) > 1e-12  # in case simplify didn't handle it
+
+
+def test_issue_14608():
+    a, b = symbols('a b', commutative=False)
+    x, y = symbols('x y')
+    try:
+        collect(a*b + b*a, a)
+        raise AssertionError("Collect did not raise AttributeError")
+    except AttributeError:
+        pass
```

Comment:
This could be simplified to `raises(AttributeError, lambda: collect(a*b + b*a, a))` (raises is defined in sympy.utilities.pytest).